tests/bosonic_files/test_backend_bosoniccircuit.py

```diff
@@ -0,0 +1,406 @@
+# Copyright 2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+r"""
+Unit tests for backends.bosonicbackend.bosoniccircuit.py.
+"""
+
+import numpy as np
+import strawberryfields.backends.bosonicbackend.bosoniccircuit as circuit
+import pytest
+
+pytestmark = pytest.mark.bosonic
+
+NUM_WEIGHTS_VALS = [1, 2, 5]
+NUM_MODES_VALS = [1, 3, 4]
+NUM_NEW_WEIGHTS_VALS = [1, 2, 3]
+R_VALS = np.array([-1.0, -1 / 3, 1 / 2, 1.2])
+PHI_VALS = np.array([-1.2, -1 / 5, 0.6, 1.3])
+NBAR_VALS = np.array([0, 1, 2.5])
+NUM_SHOTS_VALS = np.array([1, 2, 10])
+R_ANC_VALS = np.array([0.1, 0.8, 1.2, 5])
+ETA_ANC_VALS = np.array([0.1, 0.4, 0.9, 1])
+
+
+class TestBosonicCircuit:
+    r"""Tests all the functions inside backends.bosonicbackend.bosoniccircuit.py"""
+
+    @pytest.mark.parametrize("num_weights", NUM_WEIGHTS_VALS)
+    @pytest.mark.parametrize("num_modes", NUM_MODES_VALS)
+    def test_reset_circuit(self, num_weights, num_modes):
+        r"""Checks that the reset method instantiates correct number of modes
+        as vacuum, each with a given number of weights."""
+        # Create a state over 3 modes, 2 weights per mode
+        example = circuit.BosonicModes()
+        example.reset(3, 2)
+
+        # Confirm number of modes and weights
```

Comment:
Would suggest splitting this test case into multiple test cases for `reset`. There is a notion of "one `assert` per test case": though that might be a bit pragmatic idea in this case, it can be worth grouping similar assertions into the test cases (e.g., could create 2-3 short tests from this one).

Also, initializing tests that are similar in nature (e.g., all create a 3 modes, 2 weight bosonic circuit) could be done using pytest fixtures.

Reply:
I have split it now.